### ISSUE TITLE
`literal-comparison` (R0123) suggestion corrupts identifiers containing the substring "is"

### ISSUE THREAD

**Issue by Synvoya** [PyCQA/pylint#11146]
### Bug description

`literal-comparison` (`R0123`) builds its suggested `==` replacement by blindly `str.replace`-ing the first `is` occurrence in the rendered comparison. When the left operand is an identifier that contains the substring `is` (e.g. `axis`, `this`, `basis`, `prefix_is_set`), the suggestion is corrupted.

```python
def f():
    axis = 5
    if axis is 5:  # suggests 'ax== == 5' — should be 'axis == 5'
        return axis
    return None
```

### Command used

```shell
pylint --disable=all --enable=literal-comparison f.py
```

### Pylint output

```
R0123: In 'axis is 5', use '==' when comparing constant literals not 'is' ('ax== == 5')
```

### Expected behavior

```
R0123: In 'axis is 5', use '==' when comparing constant literals not 'is' ('axis == 5')
```

### Root cause

`pylint/checkers/base/comparison_checker.py` — `_check_literal_comparison` builds `fixed_node_str` via `incorrect_node_str.replace(is_or_is_not, equal_or_not_equal)`, which replaces the first `is` substring anywhere in the rendered string, including inside the identifier.

### Pylint version

```
pylint main (2bfab5f), astroid 4.2.0b3, Python 3.12
```
